p2p/peer.py

```diff
@@ -356,6 +356,9 @@ def is_closing(self) -> bool:
                     exc_info=True,
                 )
                 return
+            except MalformedMessage as err:
+                await self.disconnect(DisconnectReason.bad_protocol)
```

Comment:
If we receive an unexpected response, do we raise a `MalformedMessage`?

Like if we do:
- connect to peer
- request header
- disconnect from peer
- connect to same peer
- request block body
- receive header... `raise MalformedMessage` ?

Most `MalformedMessage`s seem like a good reason to disconnect, but that ^ doesn't.

Reply:
No, malformed message is literally a message that we cannot parse, invalid RLP encoding, unable to decrypt, etc.  A message that when we pass it through the msg decoder, something fundamental fails.  Nothing to do with validation, or the request/response round trips.